oioioi/mp/controllers.py

```diff
@@ -139,8 +139,20 @@ def update_user_result_for_problem(self, result):
                 ):
                     best_submission = [submission, score]
 
+            try:
+                report = SubmissionReport.objects.get(
+                    submission=best_submission[0], status='ACTIVE', kind='NORMAL'
+                )
+            except SubmissionReport.DoesNotExist:
+                report = None
+
             result.score = best_submission[1]
```

Comment:
I suggest to change typo of best_submission from tuple to normal variable and use following logic
- best_submission (currently best submission)
- add new variable best_submission_score (initially None, currently best score)
This would make this part of the code much more readable.